imi/src/main/java/org/motechproject/nms/imi/service/impl/ReschedulerServiceImpl.java

```diff
@@ -19,11 +26,16 @@
     private static final Logger LOGGER = LoggerFactory.getLogger(ReschedulerServiceImpl.class);
 
     private EventRelay eventRelay;
+    private SubscriptionService subscriptionService;
+    private CallRetryDataService callRetryDataService;
 
 
     @Autowired
-    public ReschedulerServiceImpl(EventRelay eventRelay) {
+    public ReschedulerServiceImpl(EventRelay eventRelay, SubscriptionService subscriptionService,
+                                  CallRetryDataService callRetryDataService) {
         this.eventRelay = eventRelay;
+        this.subscriptionService = subscriptionService;
+        this.callRetryDataService = callRetryDataService;
     }
 
 
```

Comment:
How do you feel about moving this part to the caller? So, CdrFileServiceImpl just adds the data to an event and publishes it with a subject. That way, this end up being being a pure service class that know how to listen to specific events and then process the data.

I understand the need to raise events and distribute load but It feels a little weird that the method exists to raise an event that is then captured by another method in the same class. this is borderline nitpick though.